torchdynamo/optimizations/log_args.py

```diff
@@ -0,0 +1,61 @@
+import json
+import os
+
+import torch
+
+from torchdynamo.optimizations.python_key import python_key_normalize
+
+aten = torch.ops.aten
+
+
+class ConvArgsAnalysis(torch.fx.Interpreter):
+    """
+    Log arguments like input shape (input, bias, weights shape)
+    and options(padding/stride/kernel size/dilation/etc) for
+    aten.convolution
+    """
+
+    def __init__(self, gm: torch.fx.GraphModule):
+        super().__init__(gm)
+
+        self.nodes_conv_args = {}
+        # self.convolution_args = inspect.getfullargspec(aten.convolution.op())[0]
+        # print(self.convolution_args)
+        # self.convolution_nargs = len(self.convolution_args)
+
+    def run(self, *args):
+        run_result = super().run(*args)
+        filename = "tmp/conv_args.json"
+        os.makedirs(os.path.dirname(filename), exist_ok=True)
+        with open(filename, "w") as fd:
+            json.dump(self.nodes_conv_args, fd)
+        return run_result
+
+    def run_node(self, n: torch.fx.Node):
+        result = super().run_node(n)
+
+        if n.op == "call_function":
+            if n.target == aten.convolution:
+                args, kwargs = self.fetch_args_kwargs_from_env(n)
+                assert len(args) == 9, "aten.convolution should have 9 args"
+                conv_args = {}
+                conv_args["input"] = args[0].shape
```

Comment:
Perhaps also record `args[n].stride()`, so we can tell the difference between channels first vs last.

For completeness `args[n].dtype` might also be useful.